packages/flutter_goldens_client/lib/client.dart

```diff
@@ -117,6 +118,23 @@ class GoldensClient {
     );
   }
 
+  Future<void> _checkCanSync() async {
+    final io.ProcessResult result = await process.run(
+      <String>['git', 'status', '--porcelain'],
+      workingDirectory: repositoryRoot.path,
+    );
+    if (result.stdout.trim().isNotEmpty) {
```

Comment:
Maybe check `result.exitCode` too?

Reply:
I'm not sure we gain much by doing so given that this check is purely to give a better error message when the next command is going to fail anyway:
1. If this command returns non-zero but there's no issue doing the pull, we errored out for no reason.
2. If this command returns non-zero but there is an issue doing the pull, we'll get an error on the pull anyway.